src/services/dataStreamer.ts

```diff
@@ -13,101 +14,179 @@ export interface StreamData {
+  await handleStream(stream, outputStream, dataType, reqId);
+  logDataCount(dataType, count, reqId);
+};
+
+/**
+ * Streams page documents. Assets found on each page are tracked using `assetData`.
+ *
+ * @param outputStream The pipeline to stream output to
+ * @param pagesCursor The cursor used for page documents
+ * @param assetData A mutable mapping of asset data and its list of pages
+ * @param opts Streaming options
+ */
 const streamPages = async (
-  pipeline: Duplex,
+  outputStream: Duplex,
   pagesCursor: FindCursor<PageDocType>,
-  req: Request,
+  assetData: Record<string, Set<string>>,
   opts: DataStreamOptions = {}
 ) => {
-  const assetData: Record<string, Set<string>> = {};
-  let pageCount = 0;
   const { updatedAssetsOnly, reqTimestamp, reqId } = opts;
-  const pagesStream = pagesCursor.stream({
-    transform(doc: PageDocType) {
-      // Grab static assets for each page. 1 static asset can be used on more than
-      // 1 page. Due to legacy considerations, 1 image can also be referred by more
-      // than 1 filename. This is suboptimal and should be changed in the future
-      doc.static_assets.forEach(({ checksum, key: filename, updated_at: updatedAt }) => {
-        // Skip to next asset if the asset has not been updated
-        if (updatedAssetsOnly && !isUpdated(reqTimestamp, updatedAt)) {
-          return;
-        }
-        if (!assetData[checksum]) {
-          assetData[checksum] = new Set();
-        }
-        assetData[checksum].add(filename);
-      });
-      const newDoc: StreamData = {
-        type: 'page',
-        data: doc,
-      };
-      pageCount++;
-      return newDoc;
-    },
-  });
+  const dataType = 'page';
+  let pageCount = 0;
 
-  pagesStream.pipe(pipeline, { end: false });
-  pagesStream.once('end', async () => {
-    logger.info(createMessage(`Found ${pageCount} pages`, reqId));
-    try {
-      await streamAssets(pipeline, assetData, req, reqId);
-    } catch (err) {
-      logger.error(createMessage(`Error trying to stream assets: ${err}`));
-      pipeline.end();
-      pipeline.destroy();
-    }
-  });
-  pagesStream.once('error', (err) => {
-    logger.error(createMessage(`There was an error streaming pages: ${err}`, reqId));
-    pagesStream.destroy();
-  });
+  const transform = (doc: PageDocType) => {
+    // Grab static assets for each page. 1 static asset can be used on more than
+    // 1 page. Due to legacy considerations, 1 image can also be referred by more
+    // than 1 filename. This is suboptimal and should be changed in the future
+    doc.static_assets.forEach(({ checksum, key: filename, updated_at: updatedAt }) => {
+      // Skip to next asset if the asset has not been updated
+      if (updatedAssetsOnly && !isUpdated(reqTimestamp, updatedAt)) {
+        return;
+      }
+      if (!assetData[checksum]) {
+        assetData[checksum] = new Set();
+      }
+      assetData[checksum].add(filename);
+    });
+
+    // TODO: Remove line number objects
```

Comment:
is this still valid?

Reply:
Nope, I can remove. It was something I didn't have time to work on

Reply:
Removed. Thanks for catching!